From 5bf6712df4a4625945781cc2c524ecff725a0e74 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Sun, 20 Nov 2022 20:20:57 +0800
Subject: [PATCH 34/56] Revert "drm/nouveau: fix a use-after-free in
 nouveau_gem_prime_import_sg_table()"

This reverts commit 5d6093c49c098d86c7b136aba9922df44aeb6944.
---
 drivers/gpu/drm/nouveau/nouveau_prime.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/nouveau/nouveau_prime.c b/drivers/gpu/drm/nouveau/nouveau_prime.c
index f08bda533bd9..5f5b87f99546 100644
--- a/drivers/gpu/drm/nouveau/nouveau_prime.c
+++ b/drivers/gpu/drm/nouveau/nouveau_prime.c
@@ -89,6 +89,7 @@ struct drm_gem_object *nouveau_gem_prime_import_sg_table(struct drm_device *dev,
 	ret = nouveau_bo_init(nvbo, size, align, NOUVEAU_GEM_DOMAIN_GART,
 			      sg, robj);
 	if (ret) {
+		nouveau_bo_ref(NULL, &nvbo);
 		obj = ERR_PTR(ret);
 		goto unlock;
 	}
-- 
2.25.1

